sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -150,14 +153,30 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
     assert(ipv6_hdr != NULL);
     memcpy(&remote->addr, &ipv6_hdr->src, sizeof(ipv6_addr_t));
     remote->family = AF_INET6;
+#if IS_USED(MODULE_SOCK_AUX_LOCAL)
+    if (aux.local != NULL) {
+        memcpy(&aux.local->addr, &ipv6_hdr->dst, sizeof(ipv6_addr_t));
+        aux.local->family = AF_INET6;
+    }
+#endif /* MODULE_SOCK_AUX_LOCAL */
     netif = gnrc_pktsnip_search_type(pkt, GNRC_NETTYPE_NETIF);
     if (netif == NULL) {
         remote->netif = SOCK_ADDR_ANY_NETIF;
+#if IS_USED(MODULE_SOCK_AUX_TIMESTAMP)
+            if (aux.timestamp != NULL) {
+            *aux.timestamp = 0;
+        }
+#endif /* MODULE_SOCK_AUX_TIMESTAMP */
     }
     else {
         gnrc_netif_hdr_t *netif_hdr = netif->data;
         /* TODO: use API in #5511 */
         remote->netif = (uint16_t)netif_hdr->if_pid;
+#if IS_USED(MODULE_SOCK_AUX_TIMESTAMP)
+        if (aux.timestamp != NULL) {
+            *aux.timestamp = gnrc_netif_hdr_get_timestamp(netif_hdr);
```

Comment:
Maybe I missed it, but how do you ensure the timestamp is set? GNRC works asynchronously, so it might take a while from the application's thread perspective until this is set.

Reply:
The timestamp is to be set e.g. by `gnrc_netif_ethernet.c` in `_recv()` at pretty much the same time the Ethernet header (of other L2 header) is set. So by the time `gnrc_sock_recv()` is able to retrieve the IPv6 header (which is done a few lines above), it should be safe to assume that the timestamp is also written.

Reply:
Arghs, sorry, I somehow thought this was send ^^

Reply:
Yeah, the sending part is indeed more tricky. I still haven't found the right approach for that yet.